main.cpp

```diff
@@ -91,7 +91,92 @@ good to go!
  Wait for my code review.
  */
 
+
+
+
+
 #include <iostream>
+
+struct FloatType
+{
+    float add(float lhs, float rhs )
+    {
+        return (lhs + rhs);
+    }
+    float subtract( float lhs, float rhs )
+    {
+        return (lhs - rhs);
+    }
+    float multiply( float lhs, float rhs )
+    {
+        return (lhs * rhs);
+    }
+    float divide( float lhs, float rhs )
+    {
+        if (rhs == 0)
+        {
+            std::cout << "WARNING: You are dividing by zero!" << std::endl;
+        }
+        return (lhs / rhs);
+    }
+};
+
+struct DoubleType
+{
+    double add(double lhs, double rhs )
+    {
+        return (lhs + rhs);
+    }
+    double subtract( double lhs, double rhs )
+    {
+        return (lhs - rhs);
+    }
+    double multiply( double lhs, double rhs )
+    {
+        return (lhs * rhs);
+    }
+    double divide( double lhs, double rhs )
+    {
+        if (rhs == 0)
```

Comment:
you're comparing a `double` with an `int`.  compare identical types to solve the warnings.
- [ ] revise